SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialAuthorityTest/SpatialAuthorityTest.cpp

```diff
@@ -0,0 +1,387 @@
+							{
+								FinishStep();
+							}
+							else
+							{
+								FinishTest(EFunctionalTestResult::Failed,
+									FString::Printf(TEXT("Was expecting only %d non replicated Actors, but found %d"),
+										NumNonReplicatedActorsExpected, NumNonReplicatedActorsInLevel));
+							}
+						}
+					}
+				});
+
+		// Destroy to be able to re-run.
+		AddStep(TEXT("Non-replicated Dynamic Actor - Destroy"), FWorkerDefinition::Server(1), nullptr,
+				[this]() {
+					DynamicNonReplicatedActor->Destroy();
+					DynamicNonReplicatedActor = nullptr;
+					FinishStep();
+				});
+	}
+
+	// GameMode.
+	{
+		AddStep(TEXT("GameMode - Determine Authority by every Server"), FWorkerDefinition::AllServers, nullptr, nullptr,
+			[this](float DeltaTime){
+				// This is running very far from the start, no need to wait more time.
+				ASpatialAuthorityTestGameMode* GameMode = GetWorld()->GetAuthGameMode<ASpatialAuthorityTestGameMode>();
+				if (GameMode == nullptr)
+				{
+					FinishTest(EFunctionalTestResult::Failed, TEXT("This test requires ASpatialAuthorityTestGameMode"));
+					return;
+				}
+
+				bool bIsStateValid;
+
+				USpatialAuthorityTestActorComponent* AuthorityComponent = GameMode->AuthorityComponent;
+				// Either it's bigger than 0 and all match (in the Authoritive Server), or all equal to zero (except for replicated).
+				if (AuthorityComponent->AuthorityOnBeginPlay > 0)
+				{
+					bIsStateValid = AuthorityComponent->AuthorityOnBeginPlay == AuthorityComponent->ReplicatedAuthorityOnBeginPlay
+						   && AuthorityComponent->AuthorityOnBeginPlay == AuthorityComponent->AuthorityOnTick
+						   && AuthorityComponent->NumAuthorityGains == 1 && AuthorityComponent->NumAuthorityLosses == 0;
+				}
+				else
+				{
+					bIsStateValid = AuthorityComponent->ReplicatedAuthorityOnBeginPlay != 0 && AuthorityComponent->AuthorityOnBeginPlay == 0
+									&& AuthorityComponent->AuthorityOnTick == 0 && AuthorityComponent->NumAuthorityGains == 0
+									&& AuthorityComponent->NumAuthorityLosses == 0;
+				}
+
+				if(bIsStateValid)
+				{
+					FinishStep();
+				}
+			}, 5.0f);
+	}
+
+	// GameState.
+	{
+		AddStep(
+			TEXT("GameState - Determine Authority by every Server"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+			[this](float DeltaTime) {
+				// Again, no need to wait for more time.
+				ASpatialAuthorityTestGameState* GameState = GetWorld()->GetGameState<ASpatialAuthorityTestGameState>();
+				if (GameState == nullptr) // GameMode already checked on previous step.
+				{
+					FinishTest(EFunctionalTestResult::Failed, TEXT("This test requires ASpatialAuthorityTestGameState"));
+					return;
+				}
+
+				bool bIsStateValid;
+
+				USpatialAuthorityTestActorComponent* AuthorityComponent = GameState->AuthorityComponent;
+				// Either it's bigger than 0 and all match the GameMode Authority (in the Authoritive Server),
+				// or all equal to zero (except for replicated).
+				if (AuthorityComponent->AuthorityOnBeginPlay > 0)
```

Comment:
Likewise here, would be good to validate that exactly one worker followed this path

Reply:
Maybe checking the local worker id == GameModeAuthority?

Reply:
In theory, the checks there would already be valid. However, what I'll do is each one that goes through there will send a message to the test, and in the end we check that only 1 sent it for GameMode and for GameState. This is the only way I can think of that would catch the authority setup being broken and all servers would get authority.